python/lsst/cell_coadds/_piecewise_bounded_field.py

```diff
@@ -0,0 +1,77 @@
+# This file is part of cell_coadds.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+
+import lsst.geom as geom
+from lsst.meas.algorithms import CoaddBoundedField
+from lsst.skymap import Index2D
+
+from ._uniform_grid import UniformGrid
+
+__all__ = ("PieceWiseBoundedField",)
+
+
+class PieceWiseBoundedField(CoaddBoundedField):
```

Comment:
I still very much like the idea of being able to attach an `ApCorrMap` to stitched `Exposure` objects. If that's still a bad idea, we just need to stop inheriting from `CoaddBoundedField` and can directly define the `evaluate` method. But the logic in `_evaluate` method is something that we're going to have to do eventually, even if we were to pass an object that quacks like `ApCorrMap` directly to `ApplyApCorrTask`.

Reply:
I would be more comfortable dropping the inheritance and calling the class something like `PiecewiseApertureCorrection` to make it extra clear that it's not actually a bounded field.  But I could live with the half-broken inheritance if it makes it significantly easier to run the coadd-processing tasks on stitched cell coadds while we stand that up and test it.  I would like to make sure we remove the inheritance before we switch to processing stitched cell coadds as our main pipeline output, even if that means I need to go write a little C++ class in meas_algorithms for it.